Fix : Warlock Soul Drain will not give a second Soul Shard if victim is under "Shadowburn"

Elmsroth opened the pull request.

Elmsroth committed:
Fix : Warlock Soul Drain will not give a second Soul Shard if victim is under "Shadowburn"

MadMaxMangos commented:
:white_check_mark: [Build Mangos ZERO Server 21.17.54-wehafoxk completed](https://ci.appveyor.com/project/MaNGOS/server-9fytl/builds/35814673) (commit https://github.com/mangoszero/server/commit/d8fd8cd34b by @Elmsroth)

Elmsroth committed:
Implement SPELL_EFFECT_TELEPORT_GRAVEYARD

Ported from CMangos
https://github.com/cmangos/mangos-classic/commit/b05bb50b9e0baa38be53414ecc81e53cd9be8525

Elmsroth committed:
Fixes players not receiving Soul Shards when multiple players are using Drain Soul and/or Shadowburn

See https://github.com/cmangos/mangos-classic/commit/d46bf1b4ca03e7cb22f4ead55bbc09f56937ddb7

MadMaxMangos commented:
:white_check_mark: [Build Mangos ZERO Server 21.17.54-pwxkjecl completed](https://ci.appveyor.com/project/MaNGOS/server-9fytl/builds/35815441) (commit https://github.com/mangoszero/server/commit/dd32414813 by @Elmsroth)

MadMaxMangos commented:
:white_check_mark: [Build Mangos ZERO Server 21.17.54-okausvgy completed](https://ci.appveyor.com/project/MaNGOS/server-9fytl/builds/35815482) (commit https://github.com/mangoszero/server/commit/cf5ed9de30 by @Elmsroth)
